### ISSUE TITLE
Getting: fatal error: can not increment endIndex

### ISSUE THREAD

**Issue by t2** [artemkrachulov/AKMaskField#4]
I am getting the error: `fatal error: can not increment endIndex` when I move out of a masked field into an unmasked field. The culprit is in the `convertRange` method.

```
public func convertRange(range: Range<Int>) -> Range<String.Index> {
    
    let startIndex = advance(self.startIndex, range.startIndex)
    let endIndex = advance(startIndex, range.endIndex - range.startIndex)
    
    return Range<String.Index>(start: startIndex, end: endIndex)
}
```

My text field is basic:

```
phoneContactField.placeholder = "Enter your phone number"
phoneContactField.textAlignment = .Center
phoneContactField.font = UIFont.caFontWithSize(18)
phoneContactField.delegate = self
phoneContactField.mask = "({ddd}) {ddd}-{dddd}"
phoneContactField.maskDelegate = self
phoneContactField.maskTemplate = "(XXX) XXX-XXXX"
phoneContactField.maskShowTemplate = true
```

Any thoughts?

**Comment by artemkrachulov** [comment]
I see problem. Will fix today.

**Comment by t2** [comment]
@artemkrachulov Thanks! Let me know an I can test.

**Comment by artemkrachulov** [comment]
Now must work, but I will make new update tomorrow.

**Comment by t2** [comment]
@artemkrachulov Wasn't clear on your last comment. Are you saying that `master` has the fix?

**Comment by artemkrachulov** [comment]
Yes, please test )

**Comment by t2** [comment]
@artemkrachulov Works great once I updated to latest Swift syntax! Thanks!

**Comment by haydarKarkin** [comment]
Hi Artem,
How can I fix that bug in Swift 1.2?
Thanks

**Comment by artemkrachulov** [comment]
Can you send your field setup?
